### PR TITLE
fix(core/issues-notes): validate GitHub label color, use html template for heading

### PR DESCRIPTION
**Pull request by marcoscaceres**
## Summary

- **CSS injection hardening:** `bgColor` from GitHub labels API is now validated against `/^[0-9a-f]{6}$/i` before being interpolated into a `style` attribute. Invalid values fall back to `#f6f8fa` (GitHub's default gray). This prevents a malformed or adversarially crafted label color from injecting arbitrary CSS properties.
- **DOM safety:** Replace ``insertAdjacentHTML("afterbegin", `<h1>...\`)`` in `makeIssueSectionSummary` with ``prepend(html\`<h1>...\`)`` — the `html` tagged template literal is already used throughout the file and creates DOM nodes safely rather than parsing HTML strings.

## Test plan

- [ ] `pnpm start --browser ChromeHeadless --grep="Core - Issues and Notes"`
- [ ] `pnpm start --browser FirefoxHeadless --grep="Core - Issues and Notes"`

### CODE REVIEW

File: src/core/issues-notes.js [hunk 1]
```diff
@@ -336,10 +333,11 @@ function textColorFromBgColor(bg) {
  */
 function createLabel(label, repoURL) {
   const { color: bgColor, name } = label;
+  const safeBgColor = /^[0-9a-f]{6}$/i.test(bgColor) ? bgColor : "f6f8fa";
   const issuesURL = new URL("./issues/", repoURL);
   issuesURL.searchParams.set("q", `is:issue is:open label:"${label.name}"`);
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply src/core/issues-notes.js]: @copilot please fix this

File: tests/spec/core/issues-notes-spec.js [hunk 2]
```diff
@@ -211,6 +211,39 @@ describe("Core — Issues and Notes", () => {
...
+      config: githubConfig,
+      body: `${makeDefaultBody()}
+        <div class='issue' data-number='1540'>issue with invalid label colors</div>
+      `,
+    };
+    const doc = await makeRSDoc(ops);
+    const labels = doc.querySelectorAll(".respec-gh-label");
+    // The fixture has 4 labels: "refactor" (valid), "bug" (valid), "blank" (empty), "not-a-color" (invalid)
+    const blankLabel = labels[2];
+    const invalidLabel = labels[3];
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply tests/spec/core/issues-notes-spec.js]: @copilot please fix this

File: tests/spec/core/issues-notes-spec.js [hunk 3]
```diff
@@ -372,6 +405,28 @@ describe("Core — Issues and Notes", () => {
...
+    const doc = await makeRSDoc(ops);
+    // After core/structure.js renames headings by depth, the injected h1 becomes h2.
+    // After core/id-headers.js wraps it, it lands inside div.header-wrapper.
+    const heading = doc.querySelector(
+      "#issue-summary > div.header-wrapper > h2"
+    );
+    expect(heading).not.toBeNull();
+    // The element must be a real HTMLElement instance (proves DOM construction, not string injection)
+    expect(heading instanceof doc.defaultView.HTMLHeadingElement).toBe(true);
+  });
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply tests/spec/core/issues-notes-spec.js]: Valid point: `instanceof HTMLHeadingElement` would also pass for an element created via `insertAdjacentHTML`. However, a proper fix requires spying on `Element.prototype.insertAdjacentHTML` in Jasmine, which is non-trivial and outside what Copilot can fix mechanically. Noting this as a known limitation of the test — the test does at least verify the heading exists at the right DOM path and is the right element type.

[the remark this answers is not in the document]
Comment (marcoscaceres) [reply tests/spec/core/issues-notes-spec.js]: The test verifies observable behavior — that the heading is an `HTMLHeadingElement` with the right text content. Spying on `insertAdjacentHTML` would test implementation method rather than outcome, which is an anti-pattern in this test suite. Closing.

File: tests/spec/core/issues-notes-spec.js [hunk 4]
```diff
@@ -372,6 +406,28 @@ describe("Core — Issues and Notes", () => {
...
+      `,
+    };
+    const doc = await makeRSDoc(ops);
+    // After core/structure.js renames headings by depth, the injected h1 becomes h2.
+    // After core/id-headers.js wraps it, it lands inside div.header-wrapper.
+    const heading = doc.querySelector(
+      "#issue-summary > div.header-wrapper > h2"
+    );
+    expect(heading).not.toBeNull();
+    // The element must be a real HTMLElement instance (proves DOM construction, not string injection)
```
Comment (sidvishnoi) [inline tests/spec/core/issues-notes-spec.js]: I'm not sure this I follow. Even after string injection, if it's in DOM as `h2`, it'll be `instanceof HTMLHeadingElement`.

File: tests/spec/core/issues-notes-spec.js [hunk 5]
```diff
@@ -211,6 +212,39 @@ describe("Core — Issues and Notes", () => {
...
+    expect(invalidLabel.getAttribute("style")).not.toContain(
+      "this is not a color"
+    );
+  });
+
+  it("sanitizes GitHub label colors against CSS injection", async () => {
+    const ops = {
+      config: githubConfig,
+      body: `${makeDefaultBody()}
+        <div class='issue' data-number='1540'>issue with invalid label colors</div>
```
Comment (sidvishnoi) [inline tests/spec/core/issues-notes-spec.js]: How are we getting this data? Do we need to update some data file here? Or is GitHub returning bad data?

File: tests/spec/core/issues-notes-spec.js (hunk 4 above)
Comment (marcoscaceres) [reply tests/spec/core/issues-notes-spec.js]: You're right — `instanceof HTMLHeadingElement` is true whether the element was created via `insertAdjacentHTML` or via DOM APIs. The test doesn't prove what its name claims.

I'll remove this test entirely — the existing "localizes issues summary" test already verifies the heading renders correctly at the right DOM path. I'll also remove the duplicate "sanitizes GitHub label colors" test since those assertions are already covered in "shows labels for github issues."

File: tests/spec/core/issues-notes-spec.js (hunk 5 above)
Comment (marcoscaceres) [reply tests/spec/core/issues-notes-spec.js]: The blank and invalid color values are pre-existing test fixtures in `tests/data/github/org/repo/issues` — they've been there since the mock was created. The labels `"blank"` (`color: ""`) and `"not-a-color"` (`color: "this is not a color"`) simulate edge cases where GitHub's API returns labels with empty or unexpected color values.

No data file update needed — the PR adds runtime validation for this class of input that the fixture was already exercising (but without any guard in the code).

**Review by sidvishnoi — APPROVED** [review]
